### PR TITLE
cdc/codec: fix nil value in canal codec.

### LINKED ISSUE #4736: canal_json don't support nil value for uint64
### What did you do?

1. Setup kafka inetgration test
2. Create table t1(id int primary key, c1 unsigned int);
3. insert into t1(id) values(1);

### What did you expect to see?

kafka sink work normally

### What did you see instead?

Error like:
[2022/03/01 18:27:11.489 +08:00] [PANIC] [canal.go:167] [“unsigned value not in type uint64”] [column="{“name”:“mch_id”,“type”:8,“flag”:193,“value”:null}"]

**Detail**:
trigger senarios:
column define default null at creating + insert without explicit column

https://github.com/pingcap/tiflow/blob/9fdebd6cab82ed32f406485474505f59e56a7e5e/cdc/entry/mounter.go#L526
image

https://github.com/pingcap/tiflow/blob/9fdebd6cab82ed32f406485474505f59e56a7e5e/cdc/sink/codec/canal.go#L166
image

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
v5.4.0
```

Upstream TiKV version (execute `tikv-server --version`):

```console
v5.4.0
```

TiCDC version (execute `cdc version`):

```console
v5.4.0
```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #4736 

### What is changed and how it works?

check nil before use the `Column.Value` in Canal 

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### Release note 

```release-note
fix `Canal-JSON` meet `unsigned` SQL Type and nullable, which cause CDC server panic.
```

### CODE REVIEW

**Review by liuzix — APPROVED** [review]

File: cdc/sink/codec/canal.go [hunk 1]
```diff
@@ -162,6 +162,9 @@ func getJavaSQLType(c *model.Column, mysqlType string) (result JavaSQLType, err
 	// for **unsigned** integral types, type would be `uint64` or `string`. see reference:
 	// https://github.com/pingcap/tiflow/blob/1e3dd155049417e3fd7bf9b0a0c7b08723b33791/cdc/entry/mounter.go#L501
 	// https://github.com/pingcap/tidb/blob/6495a5a116a016a3e077d181b8c8ad81f76ac31b/types/datum.go#L423-L455
+	if c.Value == nil {
```
Comment (maxshuang) [inline cdc/sink/codec/canal.go:165]: I notice v5.4.0 make some uint64 convertion. 
Can you simplely  explain why other branches don't need this convertion?

File: cdc/sink/codec/canal_flat_test.go [hunk 2]
```diff
@@ -118,6 +118,12 @@ func (s *canalFlatSuite) TestNewCanalFlatMessage4DML(c *check.C) {
 			continue
 		}
 
+		// for `Column.Value` is nil, which mean's it is nullable, set the value to `""`
+		if obtainedValue == nil {
+			c.Assert(item.expectedValue, check.Equals, "")
```
Comment (maxshuang) [inline cdc/sink/codec/canal_flat_test.go:123]: Will this empty value related to the column type？

Comment (3AceShowHand) [reply cdc/sink/codec/canal_flat_test.go:123]: no, I think all column type can be `null`.

**Pull request merged**